### PR TITLE
[SPARK-41700][CONNECT][PYTHON] Remove `FunctionBuilder`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Remove `FunctionBuilder`

### Why are the changes needed?
since we had supported almost all the functions

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
updated UT

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Cc @grundprinzip

**Review by grundprinzip — APPROVED** [review]

File: python/pyspark/sql/connect/column.py [hunk 1]
```diff
@@ -226,7 +225,7 @@ def substr(self, startPos: Union[int, "Column"], length: Union[int, "Column"]) -
         else:
             start_exp = startPos
 
-        return F.substr(self, start_exp, length_exp)
+        return substring(self, start_exp, length_exp)
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/connect/column.py:228]: Could you fix the linter failure, @zhengruifeng ?
```
annotations failed mypy checks:
python/pyspark/sql/connect/column.py:228: error: Argument 2 to "substring" has incompatible type "Column"; expected "int"  [arg-type]
python/pyspark/sql/connect/column.py:228: error: Argument 3 to "substring" has incompatible type "Column"; expected "int"  [arg-type]
```

Comment (zhengruifeng) [reply python/pyspark/sql/connect/column.py]: sure

**Comment by zhengruifeng** [comment]
thank you all, merged into master

**Pull request closed without merging**